### ISSUE TITLE
Audit prelude before 1.0.0 release.

### ISSUE THREAD

**Issue by jemc** [savi-lang/savi#164]
The `prelude` library is the one library that we won't version separately from the language when we split the libraries out in #156 - the `prelude` library defines the non-namespaced types that will be available in all programs, and will be versioned together with the language features proper.

As such we want to carefully audit the `prelude` library for the following properties:
- it should be minimal - only types which are incredibly common, require unsafe pointer use, and/or or tie into the language/compiler in a special way should be included
- the type names and method names should be clear, consistent, and coherent
- the specified behavior of each method should be clear, consistent, and coherent
- all types and methods and should be clearly documented
- all methods should be well-tested

In other words, we want to try to go ahead and eagerly make any breaking changes we may want to make later, to avoid frivolously incrementing the major language version later for something that could have been anticipated ahead of time.
